src/backend/core/api/viewsets/service.py

```
@@ -223,32 +229,44 @@ def partial_update(self, request, *args, **kwargs):
+                serializer.instance = subscription
+                serializer.apply_entitlement_configs(subscription)
+
+        # Push last. Reuse the change-detection flag set by the pre_save signal
+        # (True on create) so an unrelated edit doesn't push needlessly.
+        if service.type == "proconnect" and getattr(
+            subscription, "_proconnect_needs_sync", True
+        ):
+            # pylint: disable=import-outside-toplevel
+            from core.services.proconnect import (
+                sync_proconnect_provider_for_subscription,
             )
```

Ruff (0.16.0)

[warning] `import` should be at the top-level of a file

(PLC0415)